addons/website_sale/static/tests/tours/website_sale_buy.js

```diff
@@ -53,14 +53,25 @@ tour.register('shop_buy_product', {
             trigger: '#cart_products tr:contains("Storage Box") a.js_add_cart_json:first',
         },
         {
-            content: "set one",
+            content: "set three", 
+            // cart_update_json take some time to answer, to prevent concurrent update, we set a new value of quantity.
+            // Like that the check amount work well
             extra_trigger: '#wrap:not(:has(#cart_products tr:contains("Storage Box")))',
             trigger: '#cart_products input.js_quantity',
-            run: 'text 1',
+            run: 'text 3',
+        },
+        {
+            content: "check quantity",
+            trigger: '#cart_products input.js_quantity:propValue(3)',
+            run: function () {}, // it's a check
+        },
+        {
+            content: "check amount",
+            trigger: '#order_total span.oe_currency_value:contains("49.50")',
```

Comment:
Only this step should be enough, just adapt the amount with "set one" instead of "set three"

Reply:
`contains("16.50")` doesn't work (I have test). That why I put three to be sur that all rpc have return.
I will do what you say.

Reply:
@rdeodoo it is red
![image](https://user-images.githubusercontent.com/16716992/126314561-2fc7249c-e0d6-4e76-bceb-ef758986f0ad.png)

Reply:
Ok thank you for checking.
Since you removed the 'useless' step, it is ok like that.
Just add the comment `// wait for cart_update_json to prevent concurrent update` on this step and :+1: 
Just don't forget to add the `run // its a check` here or it will click on the amount ;-)

Reply:
And squash commits and this should be good to ship

Reply:
With qty = 3 it is green.

Reply:
Can you apply my last comment please? Your step is a check, it doesn't need to click on the total in the cart, so it is missing an empty `run` method.
You can also add the comment I adapted and mentioned, it will help people understanding the code.
Also, remove the space before `:` in commit message title, and maybe add a quick explanation in your commit message, including the step to reproduce you already wrote on the github PR :)

Reply:
Done